fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -569,7 +569,8 @@ public String getEtagValue() {
         final Date lastModifiedDate = getLastModifiedDate();
 
         if (lastModifiedDate != null) {
-            return shaHex(getPath() + lastModifiedDate.getTime());
+            return hasType(FEDORA_NON_RDF_SOURCE_DESCRIPTION) ?
```

Comment:
I think all the resources can be divided into two groups, RDF resource and Non-RDF resource. Container belongs to RDF resource. Shouldn't we have weak Etag apply for Container?